src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -96,15 +120,18 @@ public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings se
                     });
             }
 
-            _chipSelect = new ScopeData(
-                enter: () =>
-                {
-                    _controller.Write(_cs, _settings.ChipSelectLineActiveState);
-                },
-                exit: () =>
-                {
-                    _controller.Write(_cs, !(bool)_settings.ChipSelectLineActiveState);
-                });
+            if (_cs > -1)
+            {
+                _chipSelect = new ScopeData(
+                    enter: () =>
```

Comment:
can we pass in no-op instead, otherwise we have to check for null on every data transfer